File: Licenta.Server/DataLayer/Models/Project.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Licenta.Server.DataLayer.Models
{
    public class Project
    {
        public Guid ProjectId { get; set; }
 
        public string Name { get; set; }
        public string Description { get; set; }
        public string Key { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        [ForeignKey("OwnerId")]
        public Guid OwnerId { get; set; }
        public User Owner { get; set; }
        public List<Sprint> Sprints { get; set; }
        public List<Issue> Issues { get; set; }
        public List<User> Members { get; set; }
        public List<IssueStatus> IssueStatuses { get; set; }
        public List<User> Administrators { get; set; }
        public Project()
        {
            Sprints = new List<Sprint>();
            Members = new List<User>();
        }

    }
}

File: Licenta.Frontend/src/app/Models/Project.ts
import { Issue } from "./Issue";
import { IssueLabel } from "./IssueLabel";
import { Sprint } from "./Sprint";
import { User } from "./User";

export interface Project {
    projectId: string;
    name: string;
    description: string;
    startDate: Date;
    endDate: Date;
    ownerId: string;
    owner: User;
    sprints: Sprint[];
    issues: Issue[];
    members: User[];
    issueLabels: IssueLabel[];
    key: string;
}